industry_benchmarks/utils/tests/test_fix_networks.py

```diff
@@ -79,18 +124,125 @@ def test_parse_results(results, input_alchemical_network):
+        new_network = AlchemicalNetwork.from_dict(
+            json.load(
+                open(output_alchemical_network),
+                cls=JSON_HANDLER.decoder
+            )
+        )
+        default_settings = get_settings()
+        for edge in new_network.edges:
+            assert edge.protocol.settings == default_settings
+
+
+
+    def test_fix_network_cofactor_charges(self, eg5_network, eg5_results, tmp_path, capsys):
+        """Make sure we can fix networks with cofactors and charge changes."""
 
-def test_fix_network(results, input_alchemical_network, output_dir, expected_transformations, tmp_path):
-    temp_out_dir = tmp_path / output_dir
-    fix_network(results, input_alchemical_network, temp_out_dir)
-    assert os.path.isdir(temp_out_dir)
-    output_alchemical_network = temp_out_dir / 'alchemical_network.json'
-    assert output_alchemical_network.is_file()
-    new_transformations = glob.glob(f"{temp_out_dir}/transformations/*json")
-    # The direction of the transformation can switch, therefore include both
-    # directions in the expected_transformations
-    # Some of the scores of potential transformations are the same,
-    # therefore the new_transformations will not always be the same
-    assert len(new_transformations) == 12
-    for transform in new_transformations:
-        assert transform.split("/")[-1] in expected_transformations
+        temp_out_dir = tmp_path / "eg5_tranformations"
+        command = f"--input_alchem_network_file {eg5_network} --output_extra_transformations {temp_out_dir} --result_files {' '.join(eg5_results)}"
+        # TODO change the results to trigger the charge warning
+        cli_fix_network(shlex.split(command))
+        log = capsys.readouterr().out
+        # make sure the network is fixed and the message is emitted
+        assert "Planned input  no. ligands: 4" in log
+        assert "Disconnected networks which need patching: 3" in log
+        assert "Ligands in each disconnected network: [2, 1, 1]" in log
+        assert "LOG: finding additional connections to merge the broken networks:" in log
+        # load the networks and check the protocol settings for each leg
+        output_alchemical_network = temp_out_dir / 'alchemical_network.json'
+        new_network = AlchemicalNetwork.from_dict(
+            json.load(
+                open(output_alchemical_network),
+                cls=JSON_HANDLER.decoder
+            )
+        )
+        default_charge_settings = get_settings_charge_changes()
+        defaullt_settings = get_settings()
+        # make sure that charge change edges have the correct settings
+        for edge in new_network.edges:
+            charge_diff = get_alchemical_charge_difference(edge.mapping)
+            if charge_diff != 0:
+                assert edge.protocol.settings == default_charge_settings
+            else:
+                assert edge.protocol.settings == defaullt_settings
```

Comment:
Maybe add a check for the presence of the cofactor here?

Reply:
Good idea check added in [d550db5](https://github.com/OpenFreeEnergy/IndustryBenchmarks2024/pull/158/commits/d550db5308ba7070a5d95eff93f6a7f3fc504971)